merkle/hashers/tree_hasher.go

```diff
@@ -42,3 +48,47 @@ type MapHasher interface {
 	// TODO(gbelvin): Replace Size() with BitLength().
 	Size() int
 }
+
+var (
+	maxHashers = trillian.HashStrategy(len(trillian.HashStrategy_value))
+	logHashers = make([]LogHasher, maxHashers)
+	mapHashers = make([]MapHasher, maxHashers)
+)
+
+// RegisterLogHasher registers a hasher for use.
+func RegisterLogHasher(h trillian.HashStrategy, f LogHasher) {
+	if h < 0 || h >= maxHashers {
+		panic(fmt.Sprintf("RegisterLogHasher(%v) of unknown hasher", h))
```

Comment:
I think we should check for:

* hash == UNKNOWN_HASH
* hasher == nil (possibly?)
* hash previously registered (ie, warn about overrides)

Re panic or non-panic, would all callers do:

```go
if err := RegisterLogHash(h, hasher); err != nil {
  panic(err)
}
```

... or something to that effect? If yes, then I guess the panic inside Register() is appropriate.